autorest/client.go

```diff
@@ -153,7 +153,8 @@ type Client struct {
 	PollingDelay time.Duration
 
 	// PollingDuration sets the maximum polling time after which an error is returned.
-	PollingDuration time.Duration
+	// if nil, the timeout from the Context is used
+	PollingDuration *time.Duration
```

Comment:
Instead of changing the type of PollingDuration to a pointer type, I'd recommend just checking for the zero value. That will prevent a breaking change to public surface area, and protect people from having an unhelpful instant timeout.

Reply:
On another note, it's not clear to me which value this runtime should assume, should PollingDuration not be provided:
1. [`time.Duration(math.MaxInt64) // Essentially indefinite, just over 290 years`](https://play.golang.org/p/V6QGlYzVMOv)
1. `autorest.DefaultPollingDuration // A reasonable amount of time to poll for the average Azure operation.`

Maybe you could accomplish much of the functionality of making this value optional by just making it easier to specify a polling duration that is essentially indefinite?

Reply:
> Instead of changing the type of PollingDuration to a pointer type, I'd recommend just checking for the zero value. That will prevent a breaking change to public surface area, and protect people from having an unhelpful instant timeout.

It would - however I feel by making this a zero/default value this is exposing the internals behaviour of autorest to callers; rather than having this as a nil-pointer which (IMO) is clearer (but obviously would need to be made in future)?